activerecord/lib/active_record/relation/query_methods.rb

```diff
@@ -48,6 +48,34 @@ def not(opts, *rest)
         @scope
       end
 
+      # Returns a new relation with joins and where clause to identify
+      # associated relations.
+      #
+      # For example, posts that are associated to a related author:
+      #
+      #    Post.where.associated(:author)
+      #    # SELECT "posts".* FROM "posts"
+      #    # INNER JOIN "authors" ON "authors"."id" = "posts"."author_id"
+      #    # WHERE "authors"."id" IS NOT NULL
```

Comment:
@kaspth idk if i'm missing something, but is this WHERE clause necessary?

Since we're using an INNER join, all posts with a NULL author_id will not be returned

So you can just do `Post.joins(:comment)`